src/pytorch_lightning/strategies/deepspeed.py

```diff
@@ -71,9 +71,13 @@ def remove_module_hooks(model: torch.nn.Module) -> None:
 
 class LightningDeepSpeedModule(_LightningModuleWrapperBase):
     def __init__(
-        self, pl_module: Union["pl.LightningModule", _LightningPrecisionModuleWrapperBase], precision: Union[str, int]
+        self,
+        forward_module: Optional[Union["pl.LightningModule", _LightningPrecisionModuleWrapperBase]] = None,
+        pl_module: Optional[Union["pl.LightningModule", _LightningPrecisionModuleWrapperBase]] = None,
+        precision: Union[str, int] = 32,
     ) -> None:
-        super().__init__(pl_module)
+        self._validate_init_arguments(pl_module, forward_module)
```

Comment:
Beware of https://github.com/Lightning-AI/lightning/pull/14000 as it also deprecates this class (cc @rohitgr7)

Reply:
That's fine, we can adjust depending on what lands first. It will create a merge conflict that will notify us :)